clinica/pipelines/statistics_surface/_model.py

```diff
@@ -0,0 +1,728 @@
+        import inspect
+
+        json_dict = dict()
+        for attribute in inspect.getmembers(self):
+            name, value = attribute
+            if not name.startswith("_"):
+                if not inspect.ismethod(value):
+                    if hasattr(value, "to_dict"):
+                        json_dict[name] = value.to_dict(jsonable=jsonable)
+                    elif isinstance(value, np.ndarray) and jsonable:
+                        json_dict[name] = value.tolist()
+                    else:
+                        json_dict[name] = value
+        return json_dict
+
+
+class StatisticsResultsPlotter:
+    def __init__(self, output_file: PathLike, mesh: Mesh):
+        self.output_file = output_file
+        self.mesh = mesh
+        self.plotting_extension = ".png"
+        self.no_plot = {"coefficients"}  # Elements which should not be plotted
+
+    def plot(self, result: StatisticsResults, method: str):
+        plotter = self._get_plotter(method)
+        return plotter(result)
+
+    def _get_plotter(self, method: str) -> Callable[[StatisticsResults], None]:
+        if method == "nilearn_plot_surf_stat_map":
+            return self._plot_stat_maps
+        else:
+            raise NotImplementedError(f"Plotting method {method} is not implemented.")
+
+    def _plot_stat_maps(self, result: StatisticsResults):
+        from nilearn.plotting import plot_surf_stat_map
+
+        for name, res in result.to_dict(jsonable=False).items():
+            if name not in self.no_plot:
+                texture = res
+                threshold = None
+                plot_filename = (
+                    str(self.output_file) + "_" + name + self.plotting_extension
+                )
+                if isinstance(res, dict):
+                    texture = res["P"]
+                    threshold = res["thresh"]
+                cprint(msg=f"Saving plot to {plot_filename}", lvl="info")
+                plot_surf_stat_map(
+                    self.mesh,
+                    texture,
+                    threshold=threshold,
+                    output_file=plot_filename,
+                    title=name,
+                )
+
+
+class StatisticsResultsSerializer:
+    """This class is responsible for writing instances of `StatisticsResults`
+    to disk through different methods.
+
+    Attributes
+    ----------
+    output_file: Path and filename root to be used.
+    """
+
+    def __init__(self, output_file: PathLike):
+        self.output_file = output_file
+        self.json_extension = "_results.json"
+        self.json_indent = 4
+        self.mat_extension = ".mat"
+
+    def save(self, result: StatisticsResults, method: str) -> None:
+        """Save provided `StatisticsResults` to disk with provided method.
+
+        Parameters
+        ----------
+        result: StatisticsResults to be saved.
+        method: Saving method to use.
+        """
+        writer = self._get_writer(method)
+        return writer(result)
```

Comment:
`def save(...) -> None`

```suggestion
        writer(result)
```